### ISSUE TITLE
6.0: Remove all `_` prefixes from method names

### ISSUE THREAD

**Issue by LordSimal** [cakephp/cakephp#17862]
### Description

Just like https://github.com/cakephp/cakephp/issues/17375 we should also get rid of all the `_` prefixes in our method names. There are quite a few:

![image](https://github.com/user-attachments/assets/c33a84bd-9cbc-46a1-bb6d-3e6a30ee1130)

After this is done rector rules need to be created so that users - who use custom classes and overwritten methods - don't need to adjust their code as well.

### CakePHP Version

6.0

**Comment by markstory** [comment]
I think this is doable now that we're using rector more. I can help with rector rules if you'd like.

**Comment by LordSimal** [comment]
I'd say we can tackle some of the Cake 6 setup process after the 5.1 release is done.

This could easily be combined with the https://github.com/cakephp/cakephp/issues/17375 rename

**Comment by othercorey** [comment]
One issue with mass changes like this is users won't really know what they missed since it won't cause an error.

**Comment by dereuromark** [comment]
@markstory Shall we go ahead and merge those before they become conflicting?

**Comment by LordSimal** [comment]
just so i can give my 2 last cents about this:

i dont mind either way. Just wanted to make stuff a bit more consistent. If this is too much to handle related to breaking changes then lets just dont do it and stay as it is.

**Comment by dereuromark** [comment]
Are we sure that we can safely use rector here and having concrete rules per method?
Then I say this is simple enough as automation to warrant the cleanup for 6.x

**Comment by LordSimal** [comment]
https://github.com/cakephp/upgrade/blob/5.x/config/rector/sets/cakephp50.php#L107C6-L107C18
its basically this which gets filled with data from a massive array

**Comment by LordSimal** [comment]
TLDR: Too many BC changes just for consistent code style. Lets just keep it the way it is... as always

**Comment by dereuromark** [comment]
I think we should instead use rector and create the rules and then just apply those on the code base.
This way we know they work and the users can re-use them easily then for app and plugin code.

We can start with the critical API, including _defaultConfig or other often overwrote attributes and then go from there.

**Comment by LordSimal** [comment]
I'd say we can close this since all methods have now been renamed in `6.x` 🥳
